packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -474,7 +474,9 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
                     assert(this.pendingAttach.has(id), "Unexpected attach (local) channel OP");
                     this.pendingAttach.delete(id);
                 } else {
-                    assert(!this.contexts.has(id), "Unexpected attach channel OP");
+                    assert(!this.contexts.has(id), `Unexpected attach channel OP,
```

Comment:
I'd suggest replacing assert with something like that:
```
if (this.contexts.has(id)) {
   const err = {
       errorType: ContainerErrorType.dataCorruption,
       message: "...",
       local: this.contexts.get(id) instanceof LocalChannelContext,
   };
   this.closeFn(err);
   return;
}
```

Note - I'm adding this new type in PR https://github.com/microsoft/FluidFramework/pull/3900/files, but you can copy new definition / wait a bit until I merge it

Maybe we want to continue to throw error (as it makes sure we unwind stack properly without executing any more code), but I'd rather throw structured error then rely on text messages - that way grouping in telemetry is easier, we can faster see trends, etc.

Con for throwing - if someone on the stack screws up and loses this exception, then we are screwed. So I like more direct closure flow

Reply:
@vladsud, I think I'd prefer to just enhance the assert here for now.  We want to port this to 0.26/0.27 to get more data on the bug mentioned in the PR description (looks like the new error type is not ported) and also I don't think we know conclusively at this point that the symptom is data corruption (just that we saw an op for a known context ID).

Reply:
Maybe implement different fixes for release/* branches and main?
I'm fine with either staging plan, but I think we should start moving (across the board) to more structured abandonment, especially in places like that.